### ISSUE TITLE
Limit with offset results in an infinite query that can't be cancelled

### ISSUE THREAD

**Issue by gurisko** [arangodb/arangodb#9794]
## My Environment

* __ArangoDB Version__:        3.4.7
* __Storage Engine__:             MMFiles
* __Deployment Mode__:       Single Server
* __Deployment Strategy__:   Manual Start
* __Infrastructure__:               own
* __Operating System__:        Debian 9
* __Total RAM in your machine__:        32Gb
* __Disks in use__:        HDD
* __Used Package__:              Debian or Ubuntu .deb

## Component, Query & Data

__Affected feature__:  AQL

__AQL query (if applicable)__:
```
FOR d IN vDevice
    FILTER d._ci == 'vCollect/3434189746'
    FOR p IN vPortMirroring
        FILTER d._id == p.vDeviceId
        LIMIT 26, 1
        RETURN 1
```

The query works when:
* I use `LIMIT 25, 1` or any lower offset;
* I iterate over vPortMirroring first and then over vDevice.

vDevice has 7,215 documents,
vPortMirroring only has 78 records.

__AQL explain (if applicable)__:
```
Execution plan:
 Id   NodeType          Est.   Comment
  1   SingletonNode        1   * ROOT
 12   IndexNode         7215     - FOR d IN vDevice   /* hash index scan */
 11   IndexNode         7215       - FOR p IN vPortMirroring   /* hash index scan, scan only */
  8   LimitNode            1         - LIMIT 26, 1
  9   CalculationNode      1         - LET #6 = 1   /* json expression */   /* const assignment */
 10   ReturnNode           1         - RETURN #6

Indexes used:
 By   Type   Collection       Unique   Sparse   Selectivity   Fields            Ranges
 12   hash   vDevice          false    false         0.01 %   [ `_ci` ]         (d.`_ci` == "vCollect/3434189746")
 11   hash   vPortMirroring   false    false        92.31 %   [ `vDeviceId` ]   (d.`_id` == p.`vDeviceId`)

Optimization rules applied:
 Id   RuleName
  1   use-indexes
  2   remove-filter-covered-by-index
  3   remove-unnecessary-calculations-2

```

__Dataset__:
https://gist.github.com/gurisko/7e4993d65b94040a9cc2eb6052f08992

## Steps to reproduce

1. add both collections (vDevice, vPortMirroring)
2. add indexes as they are described in explain
3. run the query

__Problem__:
The query runs infinitely. It can't be cancelled (It is still visible in "Running queries" view and cancellation results in nothing). The CPU usage goes over the roof.

__Expected result__:
The query works.

**Comment by maxkernbach** [comment]
Hi @gurisko,

Thanks for reporting. I was able to reproduce this behavior with the provided data set. The query runs infinitely and can not be killed (even though response code 200 is returned executing `queries.kill`).

```
127.0.0.1:8529@i9794> queries.current()
[ 
  { 
    "id" : "294", 
    "query" : "FOR d IN vDevice\n    FILTER d._ci == 'vCollect/3434189746'\n    FOR p IN vPortMirroring\n        FILTER d._id == p.vDeviceId\n        LIMIT 26, 1\n        RETURN 1\n", 
    "bindVars" : { 
    }, 
    "started" : "2019-08-23T10:51:55Z", 
    "runTime" : 688.0560827255249, 
    "state" : "executing", 
    "stream" : false 
  } 
]

127.0.0.1:8529@i9794> queries.kill("294")
{ 
  "code" : 200 
}
```

This seems to be a bug when using `LIMIT offset, count` in a nested for loop. Using `LIMIT 26` instead of `LIMIT 26, 5` works. The query also finishes when removing the second for loop.

**Comment by jsteemann** [comment]
Fixed by PR #9819 . 
I am not sure exactly if this PR can still land in 3.4.8 or if it is going to be available from 3.4.9 onwards.

**Comment by maxkernbach** [comment]
Hi @gurisko,

Version 3.4.8, which includes a fix for this issue, has been released. The reported query now works as expected and can be killed.

Please find the latest release in the [download section](https://www.arangodb.com/download-major/).
